### DISCUSSION TITLE
Some questions about creating a new objective function

### DISCUSSION THREAD

**Discussion by Yuri-HubL** [e0404/matRad#641] [Q&A]
Dear matRad experts,

I recently read a new article that I really wanted to repeat.
https://doi.org/10.1088/1361-6560/aaf400

There is a new objective function definition involved in this article. Since I have not come into contact with relevant content before, I am a bit at a loss. Could you please give me some suggestions?
![optimization procedure](https://github.com/e0404/matRad/assets/72288469/48499ccc-54da-48cf-9118-f3885b00cc06)
![optimization procedure ](https://github.com/e0404/matRad/assets/72288469/73ef0573-e700-42b8-a9a1-9b483f10236b)

I want to calculate this new quantity (the distribution of the nanodosimetric quantity). How do I do it?

I know that the question I asked is a little broad, which may be because I am not familiar with matRad's optimization problem. I hope that I can gradually find the key to the problem through communication with you.

Looking forward to your reply.

your sincerely,
Yuri

**Comment by Yuri-HubL** [comment]
Dear matRad experts,
After looking at the code for two days, I got a rough idea of what the objective function was all about.

Specifically, I checked the contents of the following m file.
![DoseObjectives](https://github.com/e0404/matRad/assets/72288469/caf4636d-39df-4f20-8648-7861c35d0e83)
![DoseOptimizationFunction](https://github.com/e0404/matRad/assets/72288469/d8d3be29-c49f-4ab0-8919-dc955febda46)
![projections](https://github.com/e0404/matRad/assets/72288469/13f850fb-4a11-4aba-bc51-58540473970a)

After a few days of reflection, I roughly concluded that if I want to implement the nanodosimetry distribution, I should create an m file of my own under the projections folder.

However, since I am not very familiar with the particle dose calculation, do you have any relevant articles recommended to me, and suggestions on creating my own m file? Looking forward to your reply, any suggestions you may have will be very helpful to me.

your sincerely,
Yuri

**Reply by wahln** [reply]
As said in the other thread - best is to look into our Wiki to understand the dose calculation algorithms. 
https://github.com/e0404/matRad/wiki/Dose-influence-matrix-calculation#particles
Also, our matRad-functions often have one or two references listed in the code documentation describing the function.
